api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -218,6 +218,14 @@ message Permission {
     // Please refer to :ref:`this FAQ entry <faq_how_to_setup_sni>` to learn to
     // setup SNI.
     type.matcher.v3.StringMatcher requested_server_name = 9;
+
+    // A CIDR block that will be used to match the upstream IP. This field is only applicable if the
+    // request is routed by dynamic forward proxy. If this field is present and the request did not
+    // have dynamic forward proxy in its filter chain before it was processed by the RBAC filter,
+    // the matcher will evaluate the permission as `false`. Dynamic forward proxy caches the
+    // upstream ip address in filter state and the state is invalidated after each request. Both
+    // Ipv4 and Ipv6 ranges can be matched.
```

Comment:
I'm a bit hesitate to add a field to support an extension while dfp might be an exception here. Though shouldn't this be an example that we should have an upstream HTTP filter for RBAC? while I know the current upstream filter doesn't support this yet.

Reply:
Agree that this could be solved with a upstream filter. As discussed in the issue, this is a DFP specific targeted changed to solve a particular use case. We probably need to revisit this when upstream filters are more powerful.

Reply:
Thanks @lizan for reviewing

Reply:
Yeah sounds reasonable for now. Shall we name the field more explicitly, or do we expect other filter might set that filter state as well?

Reply:
@lizan I think the construct could be used in a general context when we have upstream filters in their full form. When we generalize the RBAC policy to act on ANY of the upstream ip addresses, I think the address set could be the source of truth for it. WDYT?

Reply:
Yeah I'm not crazy about this hard coding either which is effectively going to live forever. I have a vague idea that I'm going to throw out there that we might want to explore.

What if we had some type of generic matcher extension mechanism. Basically this becomes a typed_extension_config. One of the things the extension matcher gets access to when it matches is filter state. The config for the typed extension can be whatever it wants including a string name, hard coded, etc. (If you upstream the extension it needs to be more generic but if you just have an internal extension you can hard code it all.)